File: frontend/android/dynamic/settings/src/main/java/com/sample/todo/dynamic/settings/ui/SettingsBindingModule.java
package com.sample.todo.dynamic.settings.ui;

import com.sample.todo.dynamic.settings.ui.SettingsViewModel_AssistedFactory;
import dagger.Binds;
import dagger.Module;

@Module
interface SettingsBindingModule {
    @Binds
    SettingsViewModel.Factory bind(SettingsViewModel_AssistedFactory factory);
}

File: settings/src/main/java/com/sample/todo/ui/SettingsBindingModule.kt
package com.sample.todo.ui

import androidx.lifecycle.ViewModel
import com.sample.todo.di.ViewModelKey
import dagger.Binds
import dagger.Module
import dagger.multibindings.IntoMap

@Module
abstract class SettingsBindingModule {
    @Binds
    @IntoMap
    @ViewModelKey(SettingsViewModel::class)
    abstract fun bindViewModel(settingsViewModel: SettingsViewModel): ViewModel
}